packages/beta/src/__tests__/api/documentsApi.unit.spec.ts

```diff
@@ -237,4 +237,53 @@ describe('Documents unit test', () => {
       .reply(200, { items: [] });
     await client.documents.feedback.reject([{ id: 1 }]);
   });
+
+  test('document preview uri', async () => {
+    const base = (path: string): string => {
+      return `/api/v1/projects/${project}/documents${path}`;
+    };
+
+    const prev = client.documents.preview;
+    expect(prev.buildPreviewURI(1, 'image/png')).toEqual(
+      base('/preview/?documentId=1&page=0')
+    );
+    expect(prev.buildPreviewURI(4, 'application/pdf')).toEqual(
+      base('/preview/?documentId=4')
+    );
+    expect(prev.buildPreviewURI(4, 'image/png')).toEqual(
+      base('/preview/?documentId=4&page=0')
+    );
+    expect(prev.buildPreviewURI(4, 'image/png', 2)).toEqual(
+      base('/preview/?documentId=4&page=2')
+    );
+  });
+
+  test('document preview pdf', async () => {
+    nock(mockBaseUrl)
+      .get(new RegExp('/documents/preview'))
+      .query({ documentId: 1 })
+      .once()
+      .reply(200);
+    await client.documents.preview.documentAsPdf(1);
```

Comment:
Should we assert that the response is as expected here? Not just the status code, but the contents.

Reply:
I have wondered about uploading a tiny tiny pdf and compare against that ID specifically. But if that file is accidentally deleted, the integration tests for the sdk here is going to halt every new PR.

Uploading a new document for every run and then checking against that could be done. But the added time concerned me. What do you think?

Reply:
I agree, we wouldn't want a brittle situation. But in this unit test you're mocking the response, right? My `ts` is rusty. Could you add a hardcoded arraybuffer there? It doesn't need to make sense other than having whatever headers are needed to be a pdf. I think we do something similar in the tests for the document preview service itself?

Reply:
https://github.com/cognitedata/document-preview/blob/master/src/test/kotlin/com/cognite/documentpreview/controller/PreviewControllerTest.kt#L74

Reply:
Might be more hassle than it's worth tho, as you'll kind of just end up testing the mock

Reply:
Added a test for this in the integration test when a pdf is fetched